File: ueueo-framework/ueueo-blob-storing/src/main/java/com/ueueo/BlobStoring/BlobProviderSelectorExtensions.cs
using JetBrains.Annotations;

namespace Volo.Abp.BlobStoring;

public static class BlobProviderSelectorExtensions
{
    public static IBlobProvider Get<TContainer>(
        @NonNull this IBlobProviderSelector selector)
    {
        Objects.requireNonNull(selector, nameof(selector));

        return selector.Get(BlobContainerNameAttribute.GetContainerName<TContainer>());
    }
}

File: ueueo-framework/ueueo-blob-storing/src/main/java/com/ueueo/blobstoring/BlobProviderSelectorExtensions.java
package com.ueueo.blobstoring;

import org.springframework.lang.NonNull;

import java.util.Objects;

public class BlobProviderSelectorExtensions {

    public static IBlobProvider get(@NonNull IBlobProviderSelector selector, Class<?> containerType) {
        Objects.requireNonNull(selector);

        BlobContainerNameAttribute attribute = containerType.getAnnotation(BlobContainerNameAttribute.class);
        return selector.get(attribute.name());
    }
}
